lib/ide/src/controller/module.rs

```diff
@@ -4,20 +4,48 @@
 //! and uses it for synchronizing state for text and graph representations. It provides method
 //! for registering text and graph changes. If for example text represntation will be changed, there
 //! will be notifications for both text change and graph change.
-//!
-//! This module is still on WIP state, for now it contains stubs only.
 
 use crate::prelude::*;
 
+use crate::controller::FallibleResult;
+use crate::double_representation::apply_code_change_to_id_map;
+
+use ast::Ast;
+use ast::HasRepr;
+use ast::IdMap;
+use basegl::display::shape::text::text_field::TextChangedNotification;
```

Comment:
I don't like that controller is using something from view.
Can we make this type part of controller API, so always it is only view that uses controller?

Reply:
No, because this is actually a basegl structure. And basegl should know nothing about IDE views and controllers.

Reply:
Let's try moving this to data crate.